streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java

```diff
@@ -112,10 +113,42 @@ private KafkaStreamsNamedTopologyWrapper(final Collection<NamedTopology> topolog
         );
     }
 
+    /**
+     * Provides a high-level DSL for specifying the processing logic of your application and building it into an
+     * independent topology that can be executed by this {@link KafkaStreams}.
+     *
+     * @param topologyName              The name for this topology
+     * @param topologyConfigs           The properties and any config overrides for this topology
+     *
+     * @throws IllegalArgumentException if the name contains the character sequence "__"
+     */
+    public NamedTopologyBuilder newNamedTopologyBuilder(final String topologyName, final Properties topologyConfigs) {
+        if (topologyName.contains(TaskId.NAMED_TOPOLOGY_DELIMITER)) {
+            throw new IllegalArgumentException("The character sequence '__' is not allowed in a NamedTopology, please select a new name");
+        }
+        return new NamedTopologyBuilder(topologyName, applicationConfigs, topologyConfigs);
+    }
+
+    /**
+     * Returns an empty topology for full control over the graph of streams and processor nodes that define the processing
+     * logic to be executed by this {@link KafkaStreams}.
+     *
+     * @param topologyName              The name for this topology
+     * @param topologyConfigs           The properties and any config overrides for this topology
+     *
+     * @throws IllegalArgumentException if the name contains the character sequence "__"
+     */
+    public NamedTopology newNamedTopology(final String topologyName, final Properties topologyConfigs) {
```

Comment:
nit: in some classes we call it `topologyConfigs` and in some others we call it `topologyOverrides` (e.g. the NamedTopology constructor which takes `topologyConfigs` but name it as `topologyOverrides`). How about just make all of these parameters `topologyConfigs`?